pkg/resourcecreator/deployment.go

```diff
@@ -27,12 +27,20 @@ func Deployment(app *nais.Application, opts ResourceOptions) (*appsv1.Deployment
 	if err != nil {
 		return nil, err
 	}
+
+	objectMeta := app.CreateObjectMeta()
+	if val, ok := app.Annotations["kubernetes.io/change-cause"]; ok {
+		objectMeta.Annotations = map[string]string{
```

Comment:
Might want to assign rather than override here, in case `CreateObjectMeta()` decides to populate this field.